src/managers/config.py

```diff
@@ -9,6 +9,8 @@
 import re
 import textwrap
 
+from lightkube.core.exceptions import ApiError
```

Comment:
**thought**: I wonder if we should define our own `exceptions.py` with abstractions. When it comes to executing stuff on the VM or container, it's OK to have `except (subprocess.CalledProcessError, ops.pebble.ExecError)` because one of those is from the std lib, and the other is from `ops` which is present in both VM and k8s.

The issue here would be that `lightkube` is a k8s only dependency, giving us a hard time to copy and paste the code between the two charms.
What do you think about having an `exceptions.py` with the following content?
```py
from literals import SUBSTRATE

if SUBSTRATE == "k8s":
    from lightkube.core.exceptions import ApiError

    class K8sApiError(ApiError):
        pass

else:
    class K8sApiError(Exception):
        pass
```

for the copy/pastable version, or a different file altogether like we do we `literals.py`?

Reply:
+1 to go towards a separate file eventually

Reply:
Is there any issue with also making `lightkube` a dep on VM? It's not being used beyond the error type. I don't mind making a custom exception, but it doesn't seem to be a big cost.

Reply:
My preference would be to have a custom error, since in VM we could also drop the `lightkube` dependency. This custom error seems to me to be connected to the manager. I believe it is sane to import in `managers.config` from the `managers.k8s` both the Manager and the custom exceptions related to it. 

Would `manager/k8s.py` be copied and pasted between VM and K8s? Maybe we shouldn't, we could just provide two implementations that in K8s is an actual implementation, while in VM it is a no-op for the manager and the dummy definition for k8s.

Reply:
I think we should copy the file over, just to reduce the number of differences.
Again, is there any issue with a `lightkube` dependency? We'll eventually have it with 'single-charm'.

Reply:
I guess we could make the point that adding `lightkube` to VM is a step towards unifying the two codebases. I had a look at the package, and it's fairly small anyway, even including the secondary dependencies